sdks/java/core/src/test/java/org/apache/beam/sdk/testing/BigqueryMatcherTest.java

```diff
@@ -0,0 +1,186 @@
+  @Mock private Bigquery mockBigqueryClient;
+  @Mock private Bigquery.Jobs mockJobs;
+  @Mock private Bigquery.Jobs.Query mockQuery;
+  @Mock private PipelineResult mockResult;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void testBigqueryMatcherThatSucceed() throws IOException {
+    BigqueryMatcher matcher = spy(new BigqueryMatcher(
+        appName, projectId, query, "1f342f9531b4ed978419cf6d3495736a4055b3d9"));
+    doReturn(mockBigqueryClient).when(matcher).newBigqueryClient(anyString());
+    when(mockBigqueryClient.jobs()).thenReturn(mockJobs);
+
+    TableRow row = new TableRow();
+    row.set("word", "a");
+    row.set("count", 2);
+    QueryResponse response = new QueryResponse();
+    response.setRows(Lists.newArrayList(row));
+
+    when(mockJobs.query(anyString(), any(QueryRequest.class))).thenReturn(mockQuery);
+    when(mockQuery.execute()).thenReturn(response);
+
+    assertThat(mockResult, matcher);
+    verify(matcher).newBigqueryClient(eq(appName));
+    verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
+  }
+
+  @Test
+  public void testBigqueryMatcherFailsForServiecFails() throws IOException {
+    BigqueryMatcher matcher = spy(new BigqueryMatcher(
+        appName, projectId, query, "some-checksum"));
+    doReturn(mockBigqueryClient).when(matcher).newBigqueryClient(anyString());
+    when(mockBigqueryClient.jobs()).thenReturn(mockJobs);
+    when(mockJobs.query(anyString(), any(QueryRequest.class))).thenReturn(mockQuery);
+    when(mockQuery.execute()).thenThrow(new IOException());
+
+    thrown.expect(RuntimeException.class);
+    thrown.expectMessage("Failed to retrieve BigQuery data.");
+    assertThat(mockResult, matcher);
+    verify(matcher).newBigqueryClient(eq(appName));
+    verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
+  }
+
+  @Test
+  public void testBigqueryMatcherFailsForInvalidQueryResponse() throws IOException {
+    BigqueryMatcher matcher = spy(new BigqueryMatcher(
+        appName, projectId, query, "some-checksum"));
+    doReturn(mockBigqueryClient).when(matcher).newBigqueryClient(anyString());
+    when(mockBigqueryClient.jobs()).thenReturn(mockJobs);
+    when(mockJobs.query(anyString(), any(QueryRequest.class))).thenReturn(mockQuery);
+    when(mockQuery.execute()).thenReturn(null);
+
+    try {
+      assertThat(mockResult, matcher);
+    } catch (AssertionError expected) {
+      assertThat(expected.getMessage(), containsString("was (null)"));
+      verify(matcher).newBigqueryClient(eq(appName));
+      verify(mockJobs).query(eq(projectId), eq(new QueryRequest().setQuery(query)));
+      return;
+    }
+    // Note that fail throws an AssertionError which is why it is placed out here
+    // instead of inside the try-catch block.
+    fail("AssertionError is expected.");
+  }
+
+  @Test
+  public void testNewBigqueryClient() {
```

Comment:
This test is essentially just testing your libraries -- unless you have a strong reason to keep it around, suggest dropping it.